dnf5daemon-server/session.cpp

```diff
@@ -163,20 +157,26 @@ bool Session::read_all_repos() {
         return false;
     }
     repositories_status = dnfdaemon::RepoStatus::PENDING;
-    //auto & logger = base->get_logger();
-    libdnf::repo::RepoQuery enabled_repos(*base);
-    enabled_repos.filter_enabled(true);
-    enabled_repos.filter_type(libdnf::repo::Repo::Type::AVAILABLE);
-    for (auto & repo : enabled_repos) {
-        repo->set_callbacks(std::make_unique<DbusRepoCB>(*this));
-    }
 
     bool retval = true;
 
-    try {
-        base->get_repo_sack()->update_and_load_enabled_repos(false);
-    } catch (const std::runtime_error & ex) {
-        retval = false;
+    if (session_configuration_value<bool>("load_available_repo", true)) {
+        //auto & logger = base->get_logger();
+        libdnf::repo::RepoQuery enabled_repos(*base);
+        enabled_repos.filter_enabled(true);
+        enabled_repos.filter_type(libdnf::repo::Repo::Type::AVAILABLE);
+        for (auto & repo : enabled_repos) {
+            repo->set_callbacks(std::make_unique<DbusRepoCB>(*this));
+        }
+
+        try {
+            base->get_repo_sack()->update_and_load_enabled_repos(
+                session_configuration_value<bool>("load_system_repo", true));
+        } catch (const std::runtime_error & ex) {
+            retval = false;
+        }
+    } else if (session_configuration_value<bool>("load_system_repo", true)) {
```

Comment:
Can we load the `session_configuration_value<bool>("load_available_repo", true)` and `session_configuration_value<bool>("load_system_repo", true)` before the `if` block and then just use it inside? It would improve the readability of the code I'd say and not repeating the load for `load_system_repo`...

Reply:
Sure, done.